src/custom/state/swap/trade.test.ts

```diff
@@ -0,0 +1,194 @@
+import { DEFAULT_PRECISION, LONG_PRECISION } from '@src/custom/constants'
+import { basisPointsToPercent } from '@src/utils'
+import { ChainId, Fraction, Pair, Percent, Token, TokenAmount, Trade, WETH } from '@uniswap/sdk'
+import {
+  stringToCurrency,
+  _constructTradePrice,
+  _maximumAmountInExtension,
+  _minimumAmountOutExtension
+} from './extension'
+
+const WETH_MAINNET = new Token(ChainId.MAINNET, WETH[1].address, 18)
+const DAI_MAINNET = new Token(ChainId.MAINNET, '0x6b175474e89094c44da98b954eedeac495271d0f', 18)
+
+const PAIR_WETH_DAI = new Pair(
+  new TokenAmount(WETH_MAINNET, parseUnits('100000000').toString()),
+  new TokenAmount(DAI_MAINNET, parseUnits('400000000000').toString())
+)
+
+describe('Swap PRICE Quote test', () => {
+  // mocked price response for in trade
+  const MOCKED_PRICE_OUT = {
+    short: '4000',
+    long: parseUnits('4000', DAI_MAINNET.decimals).toString()
+  }
+
+  // movked price response for out trade
+  const MOCKED_PRICE_IN = {
+    short: '1',
+    long: parseUnits('1', WETH_MAINNET.decimals).toString()
+  }
+
+  const MOCKED_FEE_AMOUNT = {
+    short: '0.1',
+    long: parseUnits('0.1', WETH_MAINNET.decimals).toString()
+  }
+
+  const currencyIn = new TokenAmount(WETH_MAINNET, MOCKED_PRICE_IN.long)
+  const currencyOut = new TokenAmount(DAI_MAINNET, MOCKED_PRICE_OUT.long)
+
+  const tradeOptions = { maxHops: 1, maxNumResults: 1 }
+
+  describe('SELL', () => {
+    let tradeSdk, trade: any
+
+    describe('1 WETH --> DAI @ 0.5% slippage', () => {
+      beforeAll(() => {
+        // make a new Trade object
+        tradeSdk = Trade.bestTradeExactIn([PAIR_WETH_DAI], currencyIn, currencyOut.currency, tradeOptions)[0]
+
+        const feeAsCurrency = new TokenAmount(WETH_MAINNET, MOCKED_FEE_AMOUNT.long)
+
+        const executionPrice = _constructTradePrice({
+          sellToken: currencyIn.subtract(feeAsCurrency),
+          buyToken: currencyOut,
+          kind: 'sell',
+          price: { amount: MOCKED_PRICE_OUT.long, token: DAI_MAINNET.name || 'token' }
+        })
+
+        console.debug('EXECUTION PRICE', executionPrice?.toSignificant(18))
+
+        trade = {
+          ...tradeSdk,
+          executionPrice,
+          inputAmount: tradeSdk.inputAmount.subtract(feeAsCurrency),
+          inputAmountWithFee: tradeSdk.inputAmount.subtract(feeAsCurrency),
+          inputAmountWithoutFee: tradeSdk.inputAmount,
+          outputAmount: new TokenAmount(DAI_MAINNET, MOCKED_PRICE_OUT.long),
+          maximumAmountIn(pct: Percent) {
+            return _maximumAmountInExtension(pct, this)
+          },
+          minimumAmountOut(pct: Percent) {
+            return _minimumAmountOutExtension(pct, this)
+          },
+          fee: {
+            amount: MOCKED_FEE_AMOUNT.long,
+            feeAsCurrency
+          }
+        }
+      })
+      it('Uses proper input amount WITHOUT fee in trade object', () => {
+        // WHEN --> Sold_tokens => From_amount - Fee
```

Comment:
Much better, it's short and easier to read. But still doesn't follow the BDD. For example https://martinfowler.com/bliki/GivenWhenThen.html#:~:text=Given%2DWhen%2DThen%20is%20a,%2DDriven%20Development%20(BDD).

I'd love to be able to read this as:
```
// GIVEN: Fee is 0.1 ETH
// WHEN: User is selling 10 ETH
// THEN: The trade input after fees is 9.99 ETH
```

cc/ @alfetopito

Reply:
My comments from your previous PR they still apply, for example, we should avoid adding code to our expected values, this makes it harder to  read that setting a trivial example and showing what you expect, but most importantly you can add errors in your tests. 

Ideally, in pseudocode:
```js
// GIVEN: Fee is 0.1 ETH
const fee = '0.1'

// WHEN: User is selling 10 ETH
const trade = getTrade(Type.Sell, 10, fee);

// THEN: The trade input after fees is 9.99 ETH
expect(actualInputWithFee.raw.toString()).toEqual('9.99');
```